### PR TITLE
Abort NEST if run using mpirun but compiled without support for MPI

### LINKED ISSUE #2381: Prevent NEST from running under mpirun-control if not built with MPI
If NEST is built without MPI support, invoking NEST with `mpirun -np N` will create N independent NEST instances. This is a behaviour the user most likely does not desire. When writing to files, there is considerable risk that these instances will write to the same output files simultaneously, creating a mess at best. 

Therefore, it would be good if an MPI-free NEST could report an error if it is run from `mpirun` or similar.

Unfortunately, [this discussion on StackOverflow](https://stackoverflow.com/questions/12678337/how-can-my-program-detect-whether-it-was-launch-via-mpirun) suggest this might not be solvable in general.

I still suggest we explore and close this issue if we conclude that there is no possibility to implement this feature.

### PR DESCRIPTION
**Pull request by jougs**
This fixes #2381.

### CODE REVIEW

**Comment by jougs** [comment]
Should this be mentioned somewhere in the documentation, or do you consider this just an ordinary safeguard that goes without saying?

File: nestkernel/mpi_manager.cpp [hunk 1]
```diff
@@ -180,6 +182,19 @@ nest::MPIManager::init_mpi( int* argc, char** argv[] )
...
 nest::MPIManager::initialize()
 {
+#ifndef HAVE_MPI
+  char* pmix_rank_set = std::getenv( "PMIX_RANK" );  // set by OpenMPI's launcher
+  char* pmi_rank_set = std::getenv( "PMI_RANK" );    // set by MPICH's launcher
+  if ( pmix_rank_set or pmi_rank_set )
+  {
+    LOG( M_FATAL,
+      "MPIManager::initialize()",
+      "You seem to be using NEST via an MPI launcher like mpirun or mpiexec, "
```
Comment (heplesser) [inline nestkernel/mpi_manager.cpp:192]:
```suggestion
      "You seem to be using NEST via an MPI launcher like mpirun, mpiexec or srun "
```

**Review by heplesser — COMMENTED** [review]
Since many will be using NEST via Slurm, I think it makes sense to mention `srun` as well. I don't think we need to add more information to the documentation, since this is a corner case and the error message very informative.

But why did the "Full" test fail? I restarted the test just in case it was a random system problem.

**Comment by jougs** [comment]
> But why did the "Full" test fail? I restarted the test just in case it was a random system problem.

I have no clue. It's the same failure as in https://github.com/nest/nest-simulator/actions/runs/3550596187/jobs/5968895331, so I don't think this is related to the code here.

**Comment by med-ayssar** [comment]
As this new testing feature will not be tested by the CI, I think it is important to add an extra unittest for this case and expect it to fail.

**Review by heplesser — APPROVED** [review]
